pkg/update_pod_status_util.go

```diff
@@ -206,15 +206,17 @@ func runDD(logger logr.Logger, cmd []string, wg *sync.WaitGroup, guard chan stru
 	<-guard
 }
 
-func runBlkdiscard(logger logr.Logger, cmd []string, wg *sync.WaitGroup, guard chan struct{}) {
+func runBlkdiscard(logger logr.Logger, cmdList [][]string, wg *sync.WaitGroup, guard chan struct{}) {
 	defer wg.Done()
+	defer func() { <-guard }() // Ensure the guard channel is released
```

Comment:
add a similar defer statement for the dd command.

Reply:
yes, if panic for os.CreateTemp occurs, then defer won't run. Hence moving defer function before checking for err returned by os.CreateTemp.